File: app/src/main/java/cl/individual/calculadora/MainActivity.java
package cl.individual.calculadora;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.Editable;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button botonSumar = findViewById(R.id.botonSumar);
        Button botonRestar = findViewById(R.id.botonRestar);
        Button botonMultiplicar = findViewById(R.id.botonMultiplicar);
        Button botonDividir = findViewById(R.id.botonDividir);

        EditText numero1 = findViewById(R.id.editTextNum1);
        EditText numero2 = findViewById(R.id.editTextNum2);

        botonSumar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String num1 = numero1.getText().toString();
                int number1 = Integer.parseInt(num1);
                String num2 = numero2.getText().toString();
                int number2 = Integer.parseInt(num2);
                Toast.makeText(getBaseContext(), "The result is: " + add(number1, number2), Toast.LENGTH_LONG).show();

            }
        });

        botonRestar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String num1 = numero1.getText().toString();
                int number1 = Integer.parseInt(num1);
                String num2 = numero2.getText().toString();
                int number2 = Integer.parseInt(num2);
                Toast.makeText(getBaseContext(), "The result is: " + subtract(number1, number2), Toast.LENGTH_LONG).show();
            }
        });

        botonMultiplicar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String num1 = numero1.getText().toString();
                int number1 = Integer.parseInt(num1);
                String num2 = numero2.getText().toString();
                int number2 = Integer.parseInt(num2);
                Toast.makeText(getBaseContext(), "The result is: " + multiply(number1, number2), Toast.LENGTH_LONG).show();
            }
        });

        botonDividir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String num1 = numero1.getText().toString();
                int number1 = Integer.parseInt(num1);
                String num2 = numero2.getText().toString();
                int number2 = Integer.parseInt(num2);
                Toast.makeText(getBaseContext(), "The result is: " + divide(number1, number2), Toast.LENGTH_LONG).show();
            }
        });
    }

    public int add(int num1, int num2) {
        int result = 0;
        result = num1 + num2;
        return result;
    }

    public int subtract(int num1, int num2) {
        return num1 - num2;
    }

    public int multiply(int num1, int num2) {
        return num1 * num2;
    }

    public int divide(int num1, int num2) {
        return num1 / num2;
    }
}

File: app/src/main/java/cl/individual/calculadora/MainActivity.kt
package cl.individual.calculadora

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val botonSumar = findViewById<Button>(R.id.botonSumar)
        val botonRestar = findViewById<Button>(R.id.botonRestar)
        val botonMultiplicar = findViewById<Button>(R.id.botonMultiplicar)
        val botonDividir = findViewById<Button>(R.id.botonDividir)
        val numero1 = findViewById<EditText>(R.id.editTextNum1)
        val numero2 = findViewById<EditText>(R.id.editTextNum2)

        botonSumar.setOnClickListener {
            val num1 = numero1.text.toString()
            val num2 = numero2.text.toString()
            if (num1.trim { it <= ' ' }.isEmpty() || num2.trim { it <= ' ' }.isEmpty()) {
                Toast.makeText(baseContext, "Debe ingresar ambos numeros", Toast.LENGTH_LONG).show()
            } else {
                val number1 = num1.toInt()
                val number2 = num2.toInt()
                Toast.makeText(
                    baseContext,
                    "El resultado es:  " + add(number1, number2),
                    Toast.LENGTH_LONG
                ).show()
            }
        }
        botonRestar.setOnClickListener {
            val num1 = numero1.text.toString()
            val num2 = numero2.text.toString()
            if (num1.trim { it <= ' ' }.isEmpty() || num2.trim { it <= ' ' }.isEmpty()) {
                Toast.makeText(baseContext, "Debe ingresar ambos numeros", Toast.LENGTH_LONG).show()
            } else {
                val number1 = num1.toInt()
                val number2 = num2.toInt()
                Toast.makeText(
                    baseContext,
                    "El resultado es: " + subtract(number1, number2),
                    Toast.LENGTH_LONG
                ).show()
            }
        }
        botonMultiplicar.setOnClickListener {
            val num1 = numero1.text.toString()
            val num2 = numero2.text.toString()
            if (num1.trim { it <= ' ' }.isEmpty() || num2.trim { it <= ' ' }.isEmpty()) {
                Toast.makeText(baseContext, "Debe ingresar ambos numeros", Toast.LENGTH_LONG).show()
            } else {
                val number1 = num1.toInt()
                val number2 = num2.toInt()
                Toast.makeText(
                    baseContext,
                    "El resultado es: " + multiply(number1, number2),
                    Toast.LENGTH_LONG
                ).show()
            }
        }
        botonDividir.setOnClickListener(View.OnClickListener {
            val num1 = numero1.text.toString()
            val num2 = numero2.text.toString()
            if (num1.trim { it <= ' ' }.isEmpty() || num2.trim { it <= ' ' }.isEmpty()) {
                Toast.makeText(baseContext, "Debe ingresar ambos numeros", Toast.LENGTH_LONG).show()
                return@OnClickListener
            }
            val number1 = num1.toInt()
            val number2 = num2.toInt()
            if (number2 == 0) {
                Toast.makeText(baseContext, "Ingrese un divisor distinto de 0!", Toast.LENGTH_SHORT)
                    .show()
            } else {
                Toast.makeText(
                    baseContext,
                    "El resultado es: " + divide(number1, number2),
                    Toast.LENGTH_LONG
                ).show()
            }
        })
    }

    private fun add(num1: Int, num2: Int): Int {
        return num1 + num2
    }

    private fun subtract(num1: Int, num2: Int): Int {
        return num1 - num2
    }

    private fun multiply(num1: Int, num2: Int): Int {
        return num1 * num2
    }

    private fun divide(num1: Int, num2: Int): Int {
        return num1 / num2
    }
}